src/realm/util/file.cpp

```diff
@@ -680,21 +680,56 @@ void File::resize(SizeType size)
 }
 
 
-void File::prealloc(SizeType offset, size_t size)
+void File::prealloc(size_t size)
 {
     REALM_ASSERT_RELEASE(is_attached());
 
-#if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
-
-    prealloc_if_supported(offset, size);
-
-#else // Non-atomic fallback
+#if REALM_MACOS || defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
```

Comment:
We don't seem to have a `REALM_MACOS` and the compiler warns it evaluates to 0. I think you could add it to features.h or just use `(REALM_PLATFORM_APPLE && !REALM_MOBILE)`

Reply:
Actually, shouldn't mac go to the IOS block since like you mentioned, it doesn't have fallocate?

Reply:
true, the REALM_MACOS condition should just have been removed